### PR TITLE
HC-984 Compilation errors

### PR DESCRIPTION
**Pull request by seaxwi**
**Changes proposed in this pull request**
Expaded article on compilation errors.

### CODE REVIEW

File: content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+Learn about compilation errors in your sketch and how to resolve them.
+
+---
+
+## Quick checks
+
+* Make sure your error occurs during compilation by clicking ![Verify button](img/symbol_verify.png) **Verify** instead of ![Upload button](img/symbol_upload.png) **Upload**. This will compile the sketch without attempting to upload it. If your error only occurs when uploading, see [Errors when uploading a sketch](https://support.arduino.cc/hc/en-us/articles/4403365313810-Errors-when-uploading-a-sketch).
+
+* A successful compilation will always finish with this message:
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md:13]:
```suggestion
* A successful compilation will always finish with this message (the storage space and memory values might differ depending on the board used):
```
Just to make sure that the user don't assume that the message has to be _exactly_ the same for successful compilation

File: content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+When compiling or uploading code, this error will occur if no board is selected, or if the board core is not installed.
+
+A board needs to be selected even if you're not uploading any code.
+
+<a id="platform-not-installed"></a>
+
+### Compilation error: Error: 2 UNKNOWN: platform not installed
+
+This error occurs when the required *board core* is not installed. Please ensure the core is installed using [this guide](https://www.arduino.cc/en/Tutorial/getting-started-with-ide-v2/ide-v2-board-manager#installing-a-core).
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md:106]: The link directs to the main IDE 2 tutorials, not to the installing a core guide

File: content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+<!-- REQUIRED FUNCTIONS -->
+
+<a id="required-functions"></a>
+
+### undefined reference to 'setup' or 'loop'<br>collect2: error: ld returned 1 exit status
+
+Occurs when the `setup()` or `loop()` functions are missing. Your sketch _must_ include these functions, even if they're not being used. You can use _File > Examples > 1.Basics > BareMinium_ as a template. Note that function names are case-sensitive and that the compiler will something like `Setup()` (with a capital S) as a completely different function.
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md:114]:
```suggestion
Occurs when the `setup()` or `loop()` functions are missing. Your sketch _must_ include these functions, even if they're not being used. You can use _File > Examples > 1.Basics > BareMinimum_ as a template. Note that function names are case-sensitive and that the compiler will handle something like `Setup()` (with a capital S) as a completely different function.
```

File: content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+        <td>Error output in IDE 1.x.</td>
+        <td>Error output in IDE 2.x.</td>
+     </tr>
+   </table>
+
+---
+
+## Troubleshoot a specific error
+
+<table>
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md:59]: Missing links to anchors in some of the errors

**Review by Renat0Ribeir0 — CHANGES_REQUESTED** [review]
Good structure and images. Added some suggestions

File: content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+* Make sure your error occurs during compilation by clicking ![Verify button](img/symbol_verify.png) **Verify** instead of ![Upload button](img/symbol_upload.png) **Upload**. This will compile the sketch without attempting to upload it. If your error only occurs when uploading, see [Errors when uploading a sketch](https://support.arduino.cc/hc/en-us/articles/4403365313810-Errors-when-uploading-a-sketch).
+
+* A successful compilation will always finish with this message (the storage space and memory values might differ depending on the board used):
+
+  ```
+  Sketch uses 11604 bytes (4%) of program storage space. Maximum is 262144 bytes.
+  Global variables use 2980 bytes (9%) of dynamic memory, leaving 29788 bytes for local variables. Maximum is 32768 bytes.
+  ```
+
+* A sketch always needs to include a `setup()` and `loop()` function, even if they're not being used. You can use _File > Examples > 1.Basics > BareMinium_ as a template.
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md:20]:
```suggestion
* A sketch always needs to include a `setup()` and `loop()` function, even if they're not being used. You can use _File > Examples > 1.Basics > BareMinimum_ as a template.
```

File: content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md (hunk 2 above)
Comment (seaxwi) [reply content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md]: 6134b5e

File: content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md (hunk 4 above)
Comment (seaxwi) [reply content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md:59]: 6134b5e

**Review by Renat0Ribeir0 — APPROVED** [review]
LGTM 👍

**Pull request merged**